### PR TITLE
17 handle login

### PR DESCRIPTION
**Pull request by MohammedSallout**

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
```
Comment (haitham-akram) [inline jest.config.js:1]: export { preset: 'ts-jest',
  transform: {
    '^.+\\.(ts|tsx)?$': 'ts-jest',
    '^.+\\.(js|jsx)$': 'babel-jest',
  },}

**Review by haitham-akram — APPROVED** [review]

File: jest.config.js (hunk 1 above)
Comment (MohammedSallout) [reply jest.config.js]: Done

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

**Comment by Mu7ammadAbed** [comment]
@MohammedSallout Resolve conflicts

File: src/validation/auth/login.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { object, string } from 'yup'
+
+const loginSchema = object({
+  email: string().email().required(),
+  password: string().min(5).required()
```
Comment (Mu7ammadAbed) [inline src/validation/auth/login.ts:5]:
```suggestion
  password: string().min(8).required()
```

File: src/controllers/auth/login.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      if (result) {
+        const payload = {
+          userId,
+          userType,
+          email
+        }
+
+        const token = jwt.sign(payload, process.env.SECRET_KEY as string)
+
+        res.cookie('token', token).json({ message: 'Login successfully' })
```
Comment (Mu7ammadAbed) [inline src/controllers/auth/login.ts:28]:
```suggestion
        res.cookie('token', token, { httpOnly: true, secure: process.env.NODE_ENV === 'production' })
        .json({ message: 'Login successfully' })
```

File: src/controllers/auth/login.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+const loginUsers = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+  try {
+    const { email, password } = await loginSchema.validate(req.body)
+
+    const data = await loginQuery({ email })
+
+    if (data != null) {
+      const { userId, userType, email, password: hashedPassword } = data
+
+      const result = await bcrypt.compare(password, hashedPassword)
```
Comment (Mu7ammadAbed) [inline src/controllers/auth/login.ts:17]:
```suggestion
      const isPasswordTrue = await bcrypt.compare(password, hashedPassword)
```

File: src/queries/auth/login.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+      }
+    })
+
+    return user
+  } catch (error) {
+    throw new CustomError(500, 'Internal Server Error')
+  }
+}
+
+export default loginQuery
```
Comment (Mu7ammadAbed) [inline src/queries/auth/login.ts:21]:
```suggestion
import User from '../../models/User'
import { type IUser } from '../../interfaces/models'

const loginQuery = async (userData: { email: string }): Promise<IUser | null> => {
  const { email } = userData

  const user = await User.findOne({
    where: {
      email
    }
  })

  return user
}

export default loginQuery
```

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by AbdallahAbujazar — APPROVED** [review]

**Pull request merged**